packages/server/modules/core/tests/graphSubs.spec.js

```diff
@@ -1,593 +1,742 @@
-const gql = require( 'graphql-tag' )
-const { execute } = require( 'apollo-link' )
-const { WebSocketLink } = require( 'apollo-link-ws' )
-const { SubscriptionClient } = require( 'subscriptions-transport-ws' )
-const ws = require( 'ws' )
+const expect = require('chai').expect
+const appRoot = require('app-root-path')
+const request = require('supertest')
+const gql = require('graphql-tag')
+const { execute } = require('apollo-link')
+const { WebSocketLink } = require('apollo-link-ws')
+const { SubscriptionClient } = require('subscriptions-transport-ws')
+const ws = require('ws')
 
-const knex = require( `${appRoot}/db/knex` )
+const knex = require(`${appRoot}/db/knex`)
 
-const { createUser } = require( '../services/users' )
-const { createPersonalAccessToken } = require( '../services/tokens' )
+const { createUser } = require('../services/users')
+const { createPersonalAccessToken } = require('../services/tokens')
+const { beforeEachContext } = require(`@/test/hooks`)
 
-const { sleep, noErrors } = require( `${appRoot}/test/helpers` )
+const { sleep, noErrors } = require(`${appRoot}/test/helpers`)
 
 let addr
 let wsAddr
 let childPort = null
 
-describe( 'GraphQL API Subscriptions @gql-subscriptions', ( ) => {
+describe('GraphQL API Subscriptions @gql-subscriptions', () => {
   let userA = { name: 'd1', email: 'd.1@speckle.systems', password: 'wow8charsplease' }
   let userB = { name: 'd2', email: 'd.2@speckle.systems', password: 'wow8charsplease' }
   let userC = { name: 'd3', email: 'd.3@speckle.systems', password: 'wow8charsplease' }
   let serverProcess
 
-  const getWsClient = ( wsurl, authToken ) => {
-    const client = new SubscriptionClient( wsAddr, {
-      reconnect: true,
-      connectionParams: { headers: { Authorization: authToken } }
-    },
-    ws )
+  const getWsClient = (wsurl, authToken) => {
+    const client = new SubscriptionClient(
+      wsAddr,
+      {
+        reconnect: true,
+        connectionParams: { headers: { Authorization: authToken } }
+      },
+      ws
+    )
     return client
   }
 
-  const createSubscriptionObservable = ( wsurl, authToken, query, variables ) => {
+  const createSubscriptionObservable = (wsurl, authToken, query, variables) => {
     authToken = authToken || userA.token
-    const link = new WebSocketLink( getWsClient( wsurl, authToken ) )
-    return execute( link, { query: query, variables: variables } )
+    const link = new WebSocketLink(getWsClient(wsurl, authToken))
+    return execute(link, { query: query, variables: variables })
   }
 
   // set up app & two basic users to ping pong permissions around
-  before( async function ( ) {
-    this.timeout( 10000 ) // we need to wait for the server to start in the child process!
+  before(async function () {
+    this.timeout(10000) // we need to wait for the server to start in the child process!
 
-    await knex.migrate.rollback( )
-    await knex.migrate.latest( )
+    await beforeEachContext()
```

Comment:
Main thing I did here - replaced rollback/latest w/ truncate, cause these were really slow (on top of being E2E tests)